# MoveToPregrasp takes in the (u,v) coordinates of a click on the Realsense camera,
# as well as some configuration such as the direction of the pregrasp. It then moves
# the robot to the pregrasp configuration, giving feedback on the way.

float64 u
float64 v

uint8 PREGRASP_DIRECTION_AUTO=0
uint8 PREGRASP_DIRECTION_HORIZONTAL=1
uint8 PREGRASP_DIRECTION_VERTICAL=2

uint8 pregrasp_direction

---
# Result

# TODO: Add more granular statuses based on the reason for failure.
uint8 STATUS_SUCCESS=0
uint8 STATUS_FAILURE=1

uint8 status
---
# Feedback
geometry_msgs/PoseStamped current_pose
builtin_interfaces/Duration navigation_time
builtin_interfaces/Duration estimated_time_remaining
float32 distance_remaining
